### PR TITLE
Blazor rework for AdminConsole

### PR DESCRIPTION
**Pull request by abergs**
Just added a set of blazor components and tried to render them under a new path "/welcome". Works!

CleanShot 2023-11-22 at 10 02 37

### CODE REVIEW

File: src/AdminConsole/Pages/Organization/Overview.cshtml [hunk 1]
```diff
@@ -1,10 +1,13 @@
 @page
+@using Passwordless.AdminConsole.Components
 @model OverviewModel
 
 @{
     ViewData["Title"] = Model.Org.Name + " Applications";
 }
 
+<component type="typeof(ReleaseNotes)" render-mode="ServerPrerendered"/>
+
```
Comment (abergs) [inline src/AdminConsole/Pages/Organization/Overview.cshtml:10]: This is OK but not the prettiest. In Blazor components you just do `<ReleaseNotes />`

File: src/AdminConsole/Components/ReleaseNotes.razor [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<div class="m-4 border-2 p-3 bg-amber-50">
+    <h3>We just released v1.0.55</h3>
+    <p>You should take a look on <a href="https://github.com/passwordless/passwordless-server">Github Releases</a>!</p>
+</div>
+@code {
+    
+}
```
Comment (abergs) [inline src/AdminConsole/Components/ReleaseNotes.razor:7]: Code-behind files doesn't exist, I currently don't know the 1:1 comparison to having a ctor, get and post handler.

File: src/AdminConsole/Program.cs [hunk 3]
```diff
@@ -189,5 +194,6 @@ void RunTheApp()
     app.MapPasswordless()
         .LoginRoute?.AddEndpointFilter<LoginEndpointFilter>();
     app.MapRazorPages();
+    app.MapRazorComponents<App>();
```
Comment (abergs) [inline src/AdminConsole/Program.cs:197]: I wonder how Blazor/RazorComponents will be affected by current stack, e.g. Middlewares and Auth. Most things I expect to "just work", but we need to tweak auth-policies to cover components like we've done with Pages.

**Review by Tyrrrz — APPROVED** [review]
Looks a lot better now!

Side note: we should definitely not commit that huge auto-generated `tailwind.css` in the VCS. It should be generated automatically on build/run. I know this is unrelated to this PR specifically, but something we could address next maybe.

File: src/AdminConsole/Pages/Organization/Overview.cshtml (hunk 1 above)
Comment (Tyrrrz) [reply src/AdminConsole/Pages/Organization/Overview.cshtml:10]: @abergs can you explain what that line does? Does it specify that the component should be prerendered, as opposed to everything else?

**Comment by jonashendrickx** [comment]
image

The sidebar when in anonymous mode, or not logged in state.

File: src/AdminConsole/Components/ReleaseNotes.razor (hunk 2 above)
Comment (jonashendrickx) [reply src/AdminConsole/Components/ReleaseNotes.razor:7]: You can inject services at the top of the razor file, and then override a couple lifecycle methods

File: src/AdminConsole/Pages/Organization/Overview.cshtml (hunk 1 above)
Comment (jonashendrickx) [reply src/AdminConsole/Pages/Organization/Overview.cshtml]: @Tyrrrz Supposedly it first renders the component statically, then refreshes the component again to make it interactive.

File: src/AdminConsole/Program.cs [hunk 4]
```diff
@@ -144,6 +149,11 @@ void RunTheApp()
     services.Remove(defaultLinkGeneratorDescriptor);
     services.AddSingleton<LinkGenerator>(serviceProvider => new LinkGeneratorDecorator(serviceProvider, defaultLinkGeneratorDescriptor.ImplementationType!));
 
+    builder.Services.AddRazorComponents();
+    builder.Services.AddScoped<AuthenticationStateProvider, IdentityRevalidatingAuthenticationStateProvider>();
```
Comment (abergs) [inline src/AdminConsole/Program.cs:153]:
```suggestion
// Used for Razor/Blazor experiment
    builder.Services.AddRazorComponents();
    builder.Services.AddScoped<AuthenticationStateProvider, IdentityRevalidatingAuthenticationStateProvider>();
```

File: src/AdminConsole/Components/Account/IdentityRevalidatingAuthenticationStateProvider.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+using Microsoft.AspNetCore.Components.Server;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using Passwordless.AdminConsole.Identity;
+
+namespace Passwordless.AdminConsole.Components.Account;
+
+// This is a server-side AuthenticationStateProvider that revalidates the security stamp for the connected user
+// every 30 minutes an interactive circuit is connected.
+internal sealed class IdentityRevalidatingAuthenticationStateProvider(
```
Comment (abergs) [inline src/AdminConsole/Components/Account/IdentityRevalidatingAuthenticationStateProvider.cs:12]:
```suggestion
// Used for Blazor experiment
internal sealed class IdentityRevalidatingAuthenticationStateProvider(
```

File: src/Common/Properties/launchSettings.json
Comment (abergs) [inline src/Common/Properties/launchSettings.json:1]: This seems weird, should this be removed perhaps?

**Review by abergs — COMMENTED** [review]
Some comments, otherwise OK!

**Pull request merged**